builder/builder_test.go

```diff
@@ -202,3 +203,51 @@ func TestReceipt(t *testing.T) {
 	got := builder.Receipt(want.Fee, want.FeeUnit, want.TransactionHash, trace)
 	require.Equal(t, want, got)
 }
+
+type finisher struct {
+	i      uint64
+	min    uint64
+	cancel context.CancelFunc
+}
+
+func newFinisher(min uint64, cancel context.CancelFunc) *finisher {
+	return &finisher{
+		min:    min,
+		cancel: cancel,
+	}
+}
+
+func (f *finisher) OnBlockFinalised(_ *core.Header) {
+	if f.i < f.min {
+		f.i++
+	} else {
+		f.cancel()
+	}
+}
```

Comment:
Let's generalize this so that we can easily extend and reuse it.

Maybe something like?

``` golang
var _ builder.EventListener = (*testListener)(nil)

type testListener struct {}

func (l *testListener) WaitForBlockAndDo(blockNumber, do func())
```

`WaitForBlockAndDo` is supposed to block until builder finalizes a block for height `blockNumber` and calls `do`. So to terminate a running builder we can easily call `WaitForBlockAndDo(N, contextCancel)`

We can later on extend testListener to wait for all sorts of events, like txn inclusion for example.

Reply:
> Can we wait until we need to generalize to make these decisions

It wont be too long before we need to do that. But, fair enough.